_posts/2021-07-20-5.0.0-release.md

```diff
@@ -0,0 +1,229 @@
+* https://www.iana.org/assignments/media-types/application/vnd.apache.arrow.file
+
+We recommend ".arrow" as the IPC file format file extension and ".arrows" for
+the IPC streaming format file extension.
+
+## Arrow Flight RPC notes
+
+The Go implementation now supports custom metadata and middleware, and has
+been added to integration testing.
+
+In Python, some operations can now be interrupted via Control-C.
+
+## C++ notes
+
+`MakeArrayFromScalar` now works for fixed-size binary types (ARROW-13321).
+
+### Compute layer
+
+The following [compute functions]({{site.baseurl}}/docs/cpp/compute.html)
+were added:
+
+* aggregations: "index"
+
+* scalar arithmetic and math functions: "abs", "abs_checked", "acos",
+  "acos_checked", "asin", "asin_checked", "atan", "atan2", "ceil", "cos",
+  "cos_checked", "floor", "ln", "ln_checked", "log10", "log10_checked",
+  "log1p", "log1p_checked", "log2", "log2_checked", "negate", "negate_checked",
+  "sign", "sin", "sin_checked", "tan", "tan_checked", "trunc"
+
+* scalar bitwise functions: "bit_wise_and", "bit_wise_not", "bit_wise_or",
+  "bit_wise_xor", "shift_left", "shift_left_checked", "shift_right",
+  "shift_right_checked"
+
+* scalar string functions: "ascii_center", "ascii_lpad", "ascii_reverse",
+  "ascii_rpad", "binary_join", "binary_join_element_wise",
+  "binary_replace_slice", "count_substring", "count_substring_regex",
+  "ends_with", "find_substring", "find_substring_regex", "match_like",
+  "split_pattern_regex", "starts_with", "utf8_center", "utf8_lpad",
+  "utf8_replace_slice", "utf8_rpad", "utf8_reverse", "utf8_slice_codepoints"
+
+* scalar temporal functions: "day", "day_of_week", "day_of_year",
+  "iso_calendar", "iso_week", "iso_year", "hour", "microsecond", "millisecond",
+  "minute", "month", "nanosecond", "quarter", "second", "subsecond", "year"
+
+* other scalar functions: "case_when", "coalesce", "if_else", "is_finite",
+  "is_inf", "is_nan", "max_element_wise", "min_element_wise", "make_struct"
+
+* vector functions: "replace_with_mask"
+
+Duplicates are now allowed in `SetLookupOptions::value_set` (ARROW-12554).
+
+Decimal types are now supported by some basic arithmetic functions (ARROW-12074).
+
+The "take" function now supports dense unions (ARROW-13005).
+
+It is now possible to cast between dictionary types with different index
+types (ARROW-11673).
+
+Sorting is now implemented for boolean input (ARROW-12016).
+
+### CSV
+
+The streaming CSV reader can now read in parallel (ARROW-11889).
+
+The CSV reader tries to make its errors more informative by adding the
+row number when it is known, i.e. when parallel reading is disabled (ARROW-12675).
+
+A new option `ReaderOptions::skip_rows_after_names` allows skipping a number
+of rows _after_ reading the column names (as opposed to
+`ReaderOptions::skip_rows`).
+
+Quoted strings can now be treated as always non-null (ARROW-10115).
+
+### Dataset layer
+
```

Comment:
Someone needs to fill this. @bkietz @lidavidm

Reply:
```suggestion
The asynchronous scanner introduced in 4.0.0 has been improved, with truly asynchronous readers implemented for CSV, Parquet, and IPC file formats and file-level parallelism added.

A CountRows method has been added to count rows matching a predicate; where possible, this will use metadata in files instead of reading the data itself.

CSV datasets can now be written, and when reading a CSV dataset, explicit types can now be specified for a subset of columns while allowing the rest to still be inferred.
```

Reply:
The asynchronous scanner was never really introduced (as, explained to the users) in 4.0.0.  It seems the only reference to it in the 4.0.0 docs was [this](https://arrow.apache.org/docs/cpp/api/dataset.html#_CPPv4N5arrow7dataset11ScanOptions9use_asyncE).  Maybe we can add a bit to this.

```
The asynchronous scanner introduced in 4.0.0 has been improved, with truly asynchronous readers implemented for CSV, Parquet, and IPC file formats and file-level parallelism added.  This mode is controlled by a flag `use_async` that can be passed into methods which scan a dataset.  Setting this flag to True will have significant improvements on filesystems with high latency and generally some smaller improvements elsewhere.  This flag will default to True in some future version.
```

Reply:
@westonpace please edit the suggestion to your satisfaction, then commit it